src/templates/_layouts/components/global-sidebar.twig

```diff
@@ -101,7 +101,7 @@
                             {% if item.subnav %}
                                 <ul class="nav-item__subnav" id="{{ item.id }}-subnav" data-state="{{ item.sel ?  'expanded' : 'collapsed' }}">
                                     {% for itemId, item in item.subnav %}
-                                        {% set itemIsSelected = selectedSubnavItem is defined and selectedSubnavItem == itemId -%}
+                                        {% set itemIsSelected = selectedSubnavItem is defined and selectedSubnavItem == itemId and item.url in craft.app.request.fullPath -%}
```

Comment:
We need both conditions. The first will unintentionally match any subnav item with the same ID (usually something like "settings") while the second unintentionally matches subnav items that have similar IDs (like `commerce/inventory` and `commerce/inventory-locations`). Together they seem to only match the actually active subnav item.